internal/api/observability/secrets.go

```diff
@@ -11,3 +16,79 @@ func NewSecretReference(keyName, secretName string) *obs.SecretReference {
 		SecretName: secretName,
 	}
 }
+
+// Secrets is a map of secrets
+type Secrets map[string]*corev1.Secret
+
+// Hash64a returns an FNV-1a representation of the secrets
+func (s Secrets) Hash64a() string {
+	names := s.Names()
+	sort.Strings(names)
+	buffer := fnv.New64a()
+	//buffer := bytes.NewBufferString("")
+	for _, name := range names {
+		secret := s[name]
+		buffer.Write([]byte(name))
+
+		var keys []string
+		for key, _ := range secret.Data {
+			keys = append(keys, key)
+		}
+		sort.Strings(keys)
+
+		for _, k := range keys {
+			v := secret.Data[k]
+			buffer.Write([]byte(k))
+			buffer.Write(v)
+		}
+	}
+	return fmt.Sprintf("%d", buffer.Sum64())
```

Comment:
So far I have seen general consensus to be "output hash as hex digits", should we do this here as well?

```suggestion
	return fmt.Sprintf("%x", buffer.Sum64())
```

Reply:
For me there is value in sticking to conventions, even when it's just an implementation detail. Avoids surprises when switching from one project to another.

But ... you know ... consider it "optional". :slightly_smiling_face: